webrender/res/ps_blend.glsl

```diff
@@ -4,13 +4,29 @@
 
 #include shared,prim_shared
 
+struct ColorTransforms {
+  mat4 matrix;
+  vec4 multiplier;
+};
+
+varying ColorTransforms vColorTransforms;
 varying vec3 vUv;
 flat varying vec4 vUvBounds;
 flat varying float vAmount;
 flat varying int vOp;
 flat varying mat4 vColorMat;
 
 #ifdef WR_VERTEX_SHADER
+ColorTransforms fetch_color_transforms(int address) {
+  vec4 data[8] = fetch_from_resource_cache_8(address);
+  ColorTransforms ctransform;
+
+  ctransform.matrix = mat4(data[3], data[4], data[5], data[6]);
```

Comment:
I think it'd be clearer what's happening here if we pass in the modified address (offset), and then fetch the data from there.